### PR TITLE
Copy UserForm resource file (.frx) along with UserForm source (.frm) on export

### LINKED ISSUE #2706: Source-control gives error and removes userform from project when there's no content
I'm new to rubberduckvba.  Just installed for first time, so maybe I'm missing something easy.

Source control is removing my userforms.  It is correctly handling all other modules and class modules, but deletes forms.  Any assistance is appreciated.

Steps to reproduce:
1. create new workbook
2. open vba ide
3. insert userform
4. save xlsm workbook
5. source-control -> Manage -> initialize new repository
6. pick location and click OK
7. receive error in source-control pane: "Unknown exception. Document-type component UserForm1 does not exist. Cannot import code from repository. Document-type component UserForm1 does not exist. Cannot import code from repository. Dismiss" (note: not a typo, warning sentence is repeated twice in the app)
8. Userform has been deleted from vba project

Windows 10
Excel 2010
Rubberduckvba 2.0.12.2453

RubberduckLog.txt
2017-02-17 14:56:20.2784;INFO-2.0.11.2453;Rubberduck.App;Rubberduck version 2.0.11.2453 loading:
	Operating System: Microsoft Windows NT 6.2.9200.0 x64
	Host Product: Microsoft Office 2010 x64
	Host Version: 14.0.7177.5000
	Host Executable: EXCEL.EXE;

### LINKED ISSUE #2939: Source Control removes userforms
As in title. When trying to do anything with Source Control there is an error thrown which just states something went wrong and after processing Userform is missing

### LINKED ISSUE #2983: .frx binary is not written to local repo, and not imported even when .frx is replaced
**Sequence of actions:**
1. Opened workbook, which has userforms in it already (which all have internal code within them as well) **_Linking #2706_**
2. From Source Control window > Initialise Repository
3. Error is displayed (below). Userforms cannot be imported, and are thus not in the updated Code/Project Explorer views. **_Linking #2939, #2315_**

```
2017-04-19 13:55:10.1413;INFO-2.0.13.32288;Rubberduck.App;Rubberduck version 2.0.13.32288 loading:
	Operating System: Microsoft Windows NT 6.2.9200.0 x64
	Host Product: 2007 Microsoft Office system x86
	Host Version: 12.0.6759.5000
	Host Executable: EXCEL.EXE;
2017-04-19 13:55:36.3340;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.SourceControlViewViewModel;Initializing repo;
2017-04-19 13:55:39.0561;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.SourceControlViewViewModel;Provider changed;
2017-04-19 13:55:39.1141;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.ChangesViewViewModel;Refreshing view;
2017-04-19 13:55:42.4664;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.BranchesViewViewModel;Provider changed;
2017-04-19 13:55:42.4664;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.BranchesViewViewModel;Refreshing view;
2017-04-19 13:55:43.1584;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.SourceControlViewViewModel;Displaying Error with title 'Unknown exception.' and message 'Document-type component ReleaseForm does not exist. Cannot import code from repository.
Document-type component RelStatForm does not exist. Cannot import code from repository.
Document-type component ReleaseForm does not exist. Cannot import code from repository.
Document-type component RelStatForm does not exist. Cannot import code from repository.';
2017-04-19 13:55:43.1584;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.UnsyncedCommitsViewViewModel;Provider changed;
2017-04-19 13:55:43.1774;TRACE-2.0.13.32288;Rubberduck.UI.SourceControl.UnsyncedCommitsViewViewModel;Refreshing view;
```

**Notes:**
- There is no .frx in the local repo, only a .frm
	- I have no idea, even from the TRACE log, where the .frx is being saved.
	- This means the .frm cannot be manually imported (using the VBE's built-in import feature)
	- However, if the form was previously manually exported to an alternate folder (which results in an frm. and .frx being created), then copying the .frx to the local git repo (where there was previously only a .frm), allows THAT .frm to be manually imported. 
	- It doesn't help the git import side though, as that newly copied .frx is ignored completely, even if removed from the .gitignore
		- I assume this is internally coded in RubberDuck, and nothing to do with Git properties.

### LINKED ISSUE #2994: UserForms removed when using Indenter while parsing fails
When the Indent Current Procedure feature is used while the Rubberduck code parser is unable to complete without failing, all of the UserForms are removed from the project.

### LINKED ISSUE #3019: Parser removes a userform and re-imports it as a code module
I have installed the latest version (RD 2.0.13 hotfix).

I have a code base that I was using under RD 1.4.  This project has 1 user-form in it.

I have followed the installation wiki and installed executable as administrator, and have also run the command-line registrations (again as administrator). 

When the IDE opens, the RD button says Pending. I click the button and it seems to successfully parse the first time, such that the button says "Ready".

If I do nothing at all, but hit the button again, it successfully parses again. 

Instead of doing nothing, I open one of the code modules, add one blank space on an empty line, and hit the RD button again, the following happens:
The UserForm is removed from the Forms section of the Project Explorer, ad is re-inserted under modules.  There is metadata in the form file which gives an error on compile (although compiling it is no particularly sensible anyway as the project is now corrupted).

This is the metadata
VERSION 5#
Begin {C62A69F0-16DC-11CE-9E98-00AA00574A4F} DeltaAlgoForm
   Caption = "DeltaAlgo"
   ClientHeight = 4290
   ClientLeft = 45
   ClientTop = 330
   ClientWidth = 7200
   OleObjectBlob   =   "DeltaAlgoForm.frx":0000
   StartUpPosition = 1    'CenterOwner
End

If I open the UnitTest explorer, it correctly shows me all of the tests.  If I try to re-parse from the UnitTest explorer, the same issue happens.

Any help would be greatly appreciated.

### PR DESCRIPTION
**Pull request by shadowofsilicon**
Looking at a lot of the open issues with source control, it appeared that the underlying issue was that the UserForm resource file was simply not ending up in the same directory as the UserForm source file.

VBE does the work of the export, and we already read the contents of that source file... so, we just gotta get the resource file name and copy that file to the destination.  This fix does just that.

With my testing, I believe that this closes: #2437, closes #2706, closes #2939, closes #2983, closes #2994, and closes #3019.

It might also be the solution to: #2315, #3126, but I don't know as other functionality is broken.

### CODE REVIEW

File: Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponent.cs [hunk 1]
```diff
@@ -130,6 +131,22 @@ private void ExportUserFormModule(string path)
                                .ToArray();
             }
             File.WriteAllLines(path, code);
+
+            // LINQ hates this search, therefore, iterate the long way
+            foreach (string line in contents)
+            {
+                if (line.Contains("OleObjectBlob"))
+                {
+                    var resourceFileName = line.Trim().Split(new Char[] { '"' }).ElementAt(1);
```
Comment (Vogel612) [inline Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponent.cs]: why not `line.Split("\"")[0].Trim()`?

Comment (shadowofsilicon) [inline Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponent.cs]: I get an error when I try that: `Cannot convert from 'string' to 'char'`

**Comment by retailcoder** [comment]
This is awesome, not losing the forms is a great first step in the right direction! ..and *now* I see what you were trying to do with that `OleObjectBlob`! :+1:

**Comment by shadowofsilicon** [comment]
After more testing, I don't think this closes all the issues I referenced.  I'm looking into it now.

**Comment by retailcoder** [comment]
@shadowofsilicon ok, I'll hold off on merging then.

File: Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponent.cs [hunk 2]
```diff
@@ -130,6 +131,22 @@ private void ExportUserFormModule(string path)
...
+                    var tempName = Path.GetFileName(tempFile);
+                    var destPath = Directory.GetParent(path).FullName;
+                    if (File.Exists(Path.Combine(tempFilePath, tempFile)) && !destPath.Equals(tempFilePath))
+                    {
+                        File.Copy(Path.Combine(tempFilePath, tempName), Path.Combine(destPath, resourceFileName), overwrite:true);
```
Comment (retailcoder) [inline Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponent.cs]: Shouldn't the file be *moved* instead? It's a temp file anyway, we don't want to leave useless files around... why copy it?

Comment (shadowofsilicon) [inline Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponent.cs]: Yeah, I noticed that too this morning.

**Comment by retailcoder** [comment]
Still WIP or good to go?

**Comment by shadowofsilicon** [comment]
Good to go

**Pull request merged**